doc/releases/changelog-dev.md

```diff
@@ -12,5 +12,17 @@
 
 <h3>Contributors</h3>
 
+<h3>Operator class refactor</h3>
+
+The Operator class has undergone a major refactor with the following changes:
+
+* The `matrix` representation has been modified to be a method that accepts a 
+  `wire_order` argument and calculate the correct numerical representation 
+  with respect to that ordering. Internally, `op.matrix()` calls the new static 
+  version `compute_matrix()`, which in turn derives its information from the existing 
+  `_matrix()` method which defines the "base matrix" which uses the operator's wire order.
```

Comment:
Can we remove `_matrix()`? It seems superfluous, and hard to document!

I must admit, I assumed that `compute_matrix` was the replacement for `_matrix` 🙂

Reply:
The problem is this: we want 1) a method that is easy to call and dynamic, and respects the wire order (`matrix(wire_order=None)`), 2) a method that is static, cachable and respects the wire order (`compute_matrix(..., wire_order=None)`). But then we need a place where the base matrix can be defined. That would be `_matrix`, and it should not be called, it is just a data structure.

Do you think that is too complicated? 

My worry about merging `compute_matrix` and `_matrix` is that the former contains a lot of logic to expand the matrix. Everyone defining an op would have to reproduce the logic.